ERCS/erc-7845.md

```diff
@@ -0,0 +1,644 @@
+author: Kieran Goodary (@IAmKio), Pillar Wallet (@pillarwallet), Luke Wickens (@lbw33), Rana Khoury (@RanaBug)
+discussions-to: https://ethereum-magicians.org/t/erc-7845-minimal-orchestrator-rpc/21885
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-11-07
+---
+
+## Abstract
+
+> "Hey Google, swap my SHIB for Pillar"
+
+> "Alexa, how much USDC can I buy with what's in my wallet?"
+
+> "Siri, send 10 OP to Vitalik and 5 PEPE to Deimantas"
+
+The Minimal Orchestrator RPC aims to standardise the **minimum shape and requirements** of a **request for a solution** **_from_** an arbitrary system managing an Ethereum wallet **_to_**, ultimately, an Orchestrator.
+
+An arbitrary system could be a website, device, app, server program etc - anything that manages an Ethereum wallet, **speaks Ethereum JSON-RPC** and is looking to request solutions from an Orchestrator.
+
+All solutions from an Orchestrator are ChA¹ (Chain Abstraction-first) by default.
+
+![Flow](../assets/eip-7845/flow.jpg)
+
+## Motivation
+
+Data model standards can be written in any shape. A system will often expose their external interface but require that the request to the aforementioned interface is modelled in a way that the service understands. This creates a huge level of inconsistency and in turn makes Orchestrator interoperability more difficult.
+
+Orchestrators will become more widespread and numerous over time. This is especially true with the advent of Artificial Intelligence (AI) driven systems, the continued advancement of Human Computer Interaction (HCI) devices (especially those that are voice controlled) and the emergence of Extended Reality (XR) platforms.
+
+Standardising the request object that an Orchestrator can understand from a wallet will drive adoption and make decentralised app development easier for developers that don't know how to make on-chain transactions or have the required technical understanding of block building systems.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The Orchestrator's external interface(s) that expose functionality to an end-user application or another system `MUST` use JavaScript Object Notation (JSON).
+
+The following data definitions are available and `MUST` **prefer chain abstraction (ChA¹)**, unless stated.
+
+The following sequence diagram shows the flow that this ERC aims to standardise:
+
+![Sequence Diagram](../assets/eip-7845/sequence.png)
+
+This specification follows the top level data shape of Ethereum JSON-RPC requests, as shown below:
+
+### Request
+
+The request definition is what a wallet sends to an Orchestrator for solutions to one or more problems. The request follows the specification from Ethereum JSON-RPC.
+
+#### RPC
```

Comment:
This section seems to be re-explaining JSON RPC. I'm not sure how necessary it is.

Reply:
Whilst i understand what you're suggesting - for the sake of being clear and completion, i would prefer to leave this here to make it crystal clear that this is an Ethereum RPC request and that the `params` property is a `Problem[]`.